test/lib/analyzer/os-release-detector.spec.ts

```diff
@@ -0,0 +1,233 @@
+// tslint:disable:max-line-length
+// tslint:disable:no-string-throw
+
+import * as fs from "fs";
+import * as path from "path";
+import { ExtractedLayers } from "../../../lib/extractor/types";
+
+import { detect } from "../../../lib/analyzer/os-release/static";
+
+const getOsFixtureFiles = async (
+  dir,
+  extractAction,
+): Promise<ExtractedLayers> => {
+  const root = path.join(__dirname, "../../fixtures/os", dir, "fs");
+  const fac: ExtractedLayers = {};
+
+  async function* getFiles(dir) {
+    const entries = await fs.promises.readdir(dir, { withFileTypes: true });
+    for (const entry of entries) {
+      const res = path.resolve(dir, entry.name);
+      if (entry.isDirectory()) {
+        yield* getFiles(res);
+      } else {
+        yield res;
+      }
+    }
+  }
+
+  if (fs.existsSync(root)) {
+    for await (const f of getFiles(root)) {
+      fac[f.substring(root.length)] = {
+        [extractAction]: fs.readFileSync(f).toString("utf8"),
+      };
+    }
+  }
+
+  return fac;
+};
+
+describe("os release parsing", () => {
+  it("should parse os release files correctly", async () => {
+    const examples = {
+      "alpine:2.6": {
```

Comment:
I salvaged these from the old `test/lib/analyzer/os-release-detector.test.ts` file.